x-pack/plugins/rollup/public/test/client_integration/helpers/job_create.helpers.js

```diff
@@ -34,31 +38,43 @@ export const setup = (props) => {
 
   const clickSave = () => {
     const button = testBed.find('rollupJobSaveButton');
-    button.simulate('click');
+    act(() => {
+      button.simulate('click');
+    });
     component.update();
   };
 
   // Forms
   const fillFormFields = async (step) => {
     switch (step) {
       case 'logistics':
-        form.setInputValue('rollupJobName', JOB_TO_CREATE.id);
-        await form.setInputValue('rollupIndexPattern', JOB_TO_CREATE.indexPattern, true);
-        form.setInputValue('rollupIndexName', JOB_TO_CREATE.rollupIndex);
+        act(() => {
```

Comment:
I'm not quite sure what's the difference between `await act(async () => {` and this version without `await` and `async`?

Reply:
Ideally we want to be as synchronous as possible and use `act()`. In some cases using the synchronous version is not enough because, for example, clicking a button triggers a promise that resolves on the next tick and then updates the state. The way to wait for that next tick is to use the async version `await act(async() => {...})`.

I usually start with the sync version and if the state does not update I change to the async one. With component integration tests it is not always easy to remember all the moving parts occuring after changing a field value or clicking a button.